### PR TITLE
Target/decoy parameter names; Modification comparisons

### PR DESCRIPTION
**Pull request by acesnik**

### CODE REVIEW

File: Proteomics/Modification.cs [hunk 1]
```diff
@@ -36,12 +36,14 @@ public override string ToString()
...
-            return o != null;
+            return o != null 
+                && m.id == this.id 
+                && m.modificationType == this.modificationType;
         }
 
         public override int GetHashCode()
         {
-            return 0;
+            return id.GetHashCode();
```
Comment (acesnik) [inline Proteomics/Modification.cs]: Note, this won't work if we allow null IDs. Should we disallow null for ID in the constructor?

Comment (acesnik) [inline Proteomics/Modification.cs]: We could also just return 0 here and leave it to the Equals method to iron out.

**Comment by stefanks** [comment]
What's the reason for this change?

**Comment by acesnik** [comment]
Leah's having instances where modifications with different IDs are being merged together, e.g. acetylation and http://pir0.georgetown.edu/cgi-bin/resid?id=AA0069

**Comment by acesnik** [comment]
oops -- * methylation * and http://pir0.georgetown.edu/cgi-bin/resid?id=AA0069

**Comment by acesnik** [comment]
I guess the chemical argument here is that it's possible to have two isobaric modifications with different names and the same motif, which we might want to distinguish.

**Comment by stefanks** [comment]
I guess this makes equal behave as i would expect.

**Pull request merged**